### PR TITLE
show retag tooltip use regex, reset the dialog form when retagged

### PR DESCRIPTION
**Pull request by zhoumeina**
1 add regex for retag repository and tag. 
2 reset the form value when the form is submitted. 
3 add the success notification when retag successfully. 
4 add resource files for translation. 
5 modify the validation or tag input to use the correct one.

### CODE REVIEW

File: src/portal/lib/src/image-name-input/image-name-input.component.ts [hunk 1]
```diff
@@ -16,16 +16,29 @@ export class ImageNameInputComponent implements OnInit, OnDestroy {
...
+                Validators.required,
+                Validators.pattern('^[a-z0-9]+(?:[._-][a-z0-9]+)*$')
+            ])],
+            repoName: ["", Validators.compose([
+                Validators.required,
+                Validators.pattern('^[a-z0-9]+(?:[._-][a-z0-9]+)*$')
+            ])],
+            tagName: ["", Validators.compose([
+                Validators.required,
+                Validators.pattern('^[a-zA-Z0-9_.-]+$')
```
Comment (cd1989) [inline src/portal/lib/src/image-name-input/image-name-input.component.ts]: This RE has some problems, it should be `^[\w][\w.-]{0,127}$` (Golang).

File: src/portal/lib/src/image-name-input/image-name-input.component.ts [hunk 2]
```diff
@@ -16,16 +16,29 @@ export class ImageNameInputComponent implements OnInit, OnDestroy {
...
-            repoName: ["", Validators.required],
-            tagName: ["", Validators.required],
+            projectName: ["", Validators.compose([
+                Validators.minLength(2),
+                Validators.required,
+                Validators.pattern('^[a-z0-9]+(?:[._-][a-z0-9]+)*$')
+            ])],
+            repoName: ["", Validators.compose([
+                Validators.required,
+                Validators.pattern('^[a-z0-9]+(?:[._-][a-z0-9]+)*$')
```
Comment (cd1989) [inline src/portal/lib/src/image-name-input/image-name-input.component.ts]: You can refer to the pattern in the backend for the validation, repo name also support subpath, like `aa/bb`.

https://github.com/goharbor/harbor/pull/6547/files#diff-8afa4a5b93f3fefbb482d017eaaa9f91

File: src/portal/src/i18n/lang/zh-cn-lang.json [hunk 3]
```diff
@@ -851,5 +851,11 @@
         "MSG_SUCCESS": "垃圾回收成功",
         "MSG_SCHEDULE_SET": "垃圾回收定时任务设置成功",
         "MSG_SCHEDULE_RESET": "垃圾回收定时任务已被重置"
+    },
+    "RETAG": {
+        "MSG_SUCCESS": "复制成功",
+        "TIP_REPO": "镜像仓库名被分解为路径组件。仓库名必须至少有一个小写字母、字母数字字符，可选句点、破折号或下划线分隔。严格意义上说，它必须匹配正则表达式[a-z0-9]+(?[.-][a-z0-9]+)*.如果仓库名有两个或多个路径组件，则它们必须用正斜杠('/')分隔。包括斜杠在内的仓库名的总长度必须小于256个字符。",
```
Comment (cd1989) [inline src/portal/src/i18n/lang/zh-cn-lang.json]: How about `仓库名只支持小写字母、数字以及特殊字符 [_-.]，且不能以特殊字符开头。如果包含多级路径，以 ‘/’ 分隔，且总长度不能超过 256 个字符，例如：test/busybox。`

File: src/portal/lib/src/image-name-input/image-name-input.component.ts (hunk 1 above)
Comment (zhoumeina) [reply src/portal/lib/src/image-name-input/image-name-input.component.ts]: done.

File: src/portal/lib/src/image-name-input/image-name-input.component.ts (hunk 2 above)
Comment (zhoumeina) [reply src/portal/lib/src/image-name-input/image-name-input.component.ts]: done.

File: src/portal/src/i18n/lang/zh-cn-lang.json (hunk 3 above)
Comment (zhoumeina) [reply src/portal/src/i18n/lang/zh-cn-lang.json]: done.

**Comment by cd1989** [comment]
lgtm

**Pull request merged**